### ISSUE TITLE
[med] : Many pages: Chrome + Jaws : Screen reader is not announcing the name of the pop-up dialogue box.

### ISSUE THREAD

**Issue by Stutikuls** [Expensify/App#8530]
## Action Performed:
1. Using Chrome, open URL staging.new.expensify.com
2. Navigate to Setting using Tab key and press Enter to select it.
3. Navigate to any workspace using Tab key and press using Enter to select it.
4. Navigate to More button using Tab key and press Enter on it.
5. Navigate to Delete workspace control using Tab key and press Enter on it.
6. Observe the screen reader announcement.

## Expected Result:
Screen reader should announce the name of the pop-up dialogue box as soon as it appears on the page. Screen reader should read like "Delete dialogue, are you sure you want to delete" after expanding the delete popup.

## Actual Result:
Screen reader is not announcing the name of the pop-up dialogue box. Screen reader is reading "Dialogue delete".

**Other occurrences**
Same issue also repro on #8632, #9093

## Workaround:
Yes

## Area issue was found in:
Delete popup

## Failed WCAG checkpoints
4.1.2

## User impact:
Name is the fundamental attribute for accessibility. In the absence of correct name of the UI element, the user's dependence with screen reader won't be able to interact with the UI efficiently.

## Suggested resolution:
Define a unique id on the <div> containing the popup title and use aria-labelledby on the container of the popup modal (ex., aria-labelledby to associate the title with the popup modal so it is announced when it opens. Also add role=dialogue to the popup container. 
Refer to: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

## Platform:

- Web
- Desktop App

**Version Number: **v1.1.49-1
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Notes/Photos/Videos:** Any additional supporting documentation
**Issue reported by:** Stuti 

https://user-images.githubusercontent.com/97600950/162196847-0117301f-135b-4a56-9e42-c589c917bcef.mp4
